File: check_table_structures.dart
import 'dart:io';
import 'package:supabase/supabase.dart';

Future<void> main() async {
  // Use service role to check table structures
  final supabase = SupabaseClient(
    'https://mogjjvscxjwvhtpkrlqr.supabase.co',
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im1vZ2pqdnNjeGp3dmh0cGtybHFyIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc1NzkxOTU4MCwiZXhwIjoyMDczNDk1NTgwfQ.T2ntQv-z2EL4mkGb9b3QyXM3dT8pAOFSPKvqWPd7Xoo',
  );

  print('🔍 Checking table structures...');
  
  try {
    // 1. Check tournament_participants structure by looking at actual data
    print('\n1️⃣ tournament_participants sample data:');
    final participants = await supabase
        .from('tournament_participants')
        .select('*')
        .limit(1);
    
    if (participants.isNotEmpty) {
      final sample = participants.first;
      print('Sample record structure:');
      for (var key in sample.keys) {
        print('   $key: ${sample[key]} (${sample[key].runtimeType})');
      }
    }

    // 2. Check users structure
    print('\n2️⃣ users sample data:');
    final users = await supabase
        .from('users')  
        .select('*')
        .limit(1);
    
    if (users.isNotEmpty) {
      final sample = users.first;
      print('Sample user record structure:');
      for (var key in sample.keys) {
        print('   $key: ${sample[key]} (${sample[key].runtimeType})');
      }
    }

    // 3. Check tournaments structure
    print('\n3️⃣ tournaments sample data:');
    final tournaments = await supabase
        .from('tournaments')
        .select('*')
        .limit(1);
    
    if (tournaments.isNotEmpty) {
      final sample = tournaments.first;
      print('Sample tournament record structure:');
      for (var key in sample.keys) {
        print('   $key: ${sample[key]} (${sample[key].runtimeType})');
      }
    }

  } catch (e) {
    print('❌ Error: $e');
  }

  exit(0);
}

File: check_table_structures.py
import requests
import json

# Supabase connection details with service role
SUPABASE_URL = "https://mogjjvscxjwvhtpkrlqr.supabase.co"
SERVICE_ROLE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im1vZ2pqdnNjeGp3dmh0cGtybHFyIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc1NzkxOTU4MCwiZXhwIjoyMDczNDk1NTgwfQ.T2ntQv-z2EL4mkGb9b3QyXM3dT8pAOFSPKvqWPd7Xoo"

headers = {
    'apikey': SERVICE_ROLE_KEY,
    'Authorization': f'Bearer {SERVICE_ROLE_KEY}',
    'Content-Type': 'application/json'
}

def get_table_structure(table_name):
    print(f"=== STRUCTURE OF TABLE: {table_name} ===\n")
    
    try:
        # Get first few records to understand structure
        response = requests.get(
            f"{SUPABASE_URL}/rest/v1/{table_name}",
            headers=headers,
            params={"select": "*", "limit": "3"}
        )
        
        if response.status_code == 200:
            data = response.json()
            print(f"Sample data from {table_name}:")
            print(json.dumps(data, indent=2, default=str))
            
            if data:
                print(f"\nColumns in {table_name}:")
                for key in data[0].keys():
                    print(f"  - {key}")
        else:
            print(f"Error getting {table_name}: {response.status_code} - {response.text}")
            
    except Exception as e:
        print(f"Error: {e}")

def check_user_tables():
    print("=== CHECKING USER-RELATED TABLES ===\n")
    
    user_tables = ['users', 'user_profiles', 'profiles']
    
    for table in user_tables:
        try:
            response = requests.get(
                f"{SUPABASE_URL}/rest/v1/{table}",
                headers=headers,
                params={"select": "*", "limit": "1"}
            )
            
            if response.status_code == 200:
                data = response.json()
                print(f"✅ Table '{table}' exists")
                if data:
                    print(f"   Columns: {list(data[0].keys())}")
                else:
                    print("   (Empty table)")
            else:
                print(f"❌ Table '{table}' does not exist - Status: {response.status_code}")
                
        except Exception as e:
            print(f"❌ Error checking table '{table}': {e}")
        print()

if __name__ == "__main__":
    get_table_structure("clubs")
    print("\n" + "="*50 + "\n")
    get_table_structure("club_members")
    print("\n" + "="*50 + "\n")
    check_user_tables()